### PR TITLE
Few updates

### PR DESCRIPTION
**Pull request by zeripath**
* Use jwt/v4 everywhere and use fork of markbates/goth that uses jwt/v4
* Use the same webauthnCredential definition in the migration
*  Adjust the comments to make them no longer a direct copy and add exists/name checks + further highlight we need to handle the errors properly.

### CODE REVIEW

**Comment by lunny** [comment]
Please rebase.

File: models/auth/webauthn.go [hunk 1]
```diff
@@ -39,6 +41,7 @@ func IsErrWebAuthnCredentialNotExist(err error) bool {
 type WebAuthnCredential struct {
 	ID              int64  `xorm:"pk autoincr"`
 	Name            string `xorm:"unique(s)"`
+	LowerName       string
```
Comment (lunny) [inline models/auth/webauthn.go:44]: So, the tag `xorm:"unique(s)"` should be in the `LowerName` field.

File: routers/web/auth/webauthn.go [hunk 2]
```diff
@@ -45,24 +45,24 @@ func WebAuthn(ctx *context.Context) {
 // WebAuthnLoginAssertion submits a WebAuthn challenge to the browser
 func WebAuthnLoginAssertion(ctx *context.Context) {
 	// Ensure user is in a WebAuthn session.
-	idSess := ctx.Session.Get("twofaUid")
-	if idSess == nil {
+	idSess, ok := ctx.Session.Get("twofaUid").(int64)
```
Comment (lunny) [inline routers/web/auth/webauthn.go:48]: It will panic if there is no session named `twofaUid`.

**Comment by 6543** [comment]
link https://github.com/go-gitea/gitea/pull/17957

File: routers/web/auth/webauthn.go (hunk 2 above)
Comment (zeripath) [reply routers/web/auth/webauthn.go:48]: https://goplay.space/#-2Z8pKmnVda

Comment (zeripath) [reply routers/web/auth/webauthn.go:48]: The use of the `x, ok := something.(otherthing)` form prevents the NPE if `something` is a `nil` `interface{}`

**Comment by zeripath** [comment]
OK i've fixed the ci and I've fixed a few more bugs with suburl and users.

**Comment by lunny** [comment]
Let's move on.

**Comment by zeripath** [comment]
Somehow all of the commits to this branch did not make it across. This is weird.

**Comment by zeripath** [comment]
I'm just gonna push these up directly to lunny/webauthn

**Pull request merged**